### PR TITLE
Handle Kraken decimals for OMS payloads

### PR DESCRIPTION
**Pull request by sann1432**
## Summary
- convert the synchronous OMS request/response workflow to use `Decimal` parsing for order inputs and context bookkeeping
- load Kraken asset metadata to quantize quantities, limits, and triggers before building order payloads
- retain decimal arithmetic through fee reconciliation and add unit coverage for Kraken lot/tick edge cases

## Testing
- pytest tests/unit/test_oms_service_precision.py

------
https://chatgpt.com/codex/tasks/task_e_68dee92ec1808321b072ccdb2b3de404

### CODE REVIEW

File: oms_service.py [hunk 1]
```diff
@@ -598,16 +712,70 @@ async def get_status(self, account_id: str, order_id: str) -> OrderStatusRespons
...
+            except HTTPException:
+                raise
+            except Exception as exc:  # pragma: no cover - defensive logging
+                logger.warning(
+                    "Precision validation failed for %s: %s", request.symbol, exc
+                )
+                snapped_qty = request.qty
+                snapped_limit = request.limit_px
+            pair_meta = _resolve_pair_metadata(request.symbol, metadata)
+
```
[the remark this answers is not in the document]
Comment (sann1432) [reply oms_service.py:744]: @codex fix comment
